lib/organisation_import/import_organisation_data.rb

```diff
@@ -4,11 +4,33 @@
 require "open-uri"
 
 class ImportOrganisationData
+  def run!
+    csv_metadata.each { |metadata| import_data(metadata) }
+  end
+
+  def self.mark_all_school_group_memberships_to_be_deleted!
+    SchoolGroupMembership.update_all(do_not_delete: false)
+  end
+
+  def self.delete_marked_school_group_memberships!
+    SchoolGroupMembership.where(do_not_delete: false).delete_all
```

Comment:
I would add a check here:
```
return if SchoolGroupMembership.where(do_not_delete: false).count > SOME_THRESHOLD
```
if we have too many school_group_memberships to delete, maybe the task did not run correctly